### PR TITLE
Add blurFocus option to disable bluring the focus of existing elements.

### PR DESCRIPTION
**Pull request by slimsag**
This change adds a `blurFocus` option that disables the focus blurring that occurs on existing elements when loading content. This can be useful if using PJAX to reload the contents of a search page, for instance.

Multiple other people requested this feature [back in 2013 when the blurring was introduced](https://github.com/defunkt/jquery-pjax/pull/300#commitcomment-3621248), and additionally so [on a fork](https://github.com/yiisoft/jquery-pjax/issues/19) of your project.

### CODE REVIEW

**Comment by mislav** [comment]
See https://github.com/defunkt/jquery-pjax/pull/378#issuecomment-42202492

I wouldn't add a configuration option. I would rather add a condition that elements should only get blurred/refocused if inside the pjax container.

**Comment by slimsag** [comment]
I've addressed your feedback in 70e30e5.

Thank you for your feedback and time spent on this @mislav, really appreciate it!

File: jquery.pjax.js [hunk 1]
```diff
@@ -311,7 +317,7 @@ function pjax(options) {
     //
     // http://www.w3.org/html/wg/drafts/html/master/forms.html
     var autofocusEl = context.find('input[autofocus], textarea[autofocus]').last()[0]
```
Comment (staabm) [inline jquery.pjax.js]: Shouldnt we also defer this DOM operation to only evaluate when `blurFocus` is true?

Comment (slimsag) [inline jquery.pjax.js]: Good point. Done.

File: jquery.pjax.js [hunk 2]
```diff
@@ -310,9 +316,11 @@ function pjax(options) {
     // the last field.
     //
     // http://www.w3.org/html/wg/drafts/html/master/forms.html
-    var autofocusEl = context.find('input[autofocus], textarea[autofocus]').last()[0]
-    if (autofocusEl && document.activeElement !== autofocusEl) {
-      autofocusEl.focus();
+    if(blurFocus) {
+      var autofocusEl = context.find('input[autofocus], textarea[autofocus]').last()[0]
+      if (autofocusEl && document.activeElement !== autofocusEl) {
```
Comment (staabm) [inline jquery.pjax.js]: Hmm maybe we should also defer the DOM operation until `document.activeElement` is truethy?

File: jquery.pjax.js [hunk 3]
```diff
@@ -310,9 +316,11 @@ function pjax(options) {
     // the last field.
     //
     // http://www.w3.org/html/wg/drafts/html/master/forms.html
-    var autofocusEl = context.find('input[autofocus], textarea[autofocus]').last()[0]
-    if (autofocusEl && document.activeElement !== autofocusEl) {
-      autofocusEl.focus();
+    if(blurFocus) {
```
Comment (mislav) [inline jquery.pjax.js]: If an autofocus element exists within the new pjax content, it should be autofocused unconditionally. This is the old behavior, and I don't think it should be changed. I wouldn't add this conditional here.

File: jquery.pjax.js [hunk 4]
```diff
@@ -292,10 +292,16 @@ function pjax(options) {
       window.history.replaceState(pjax.state, container.title, container.url)
     }
 
+    // Only blur the focus if the focused element is within the container.
+    var blurFocus = $.contains(options.container, $(':focus'));
```
Comment (mislav) [inline jquery.pjax.js]: 1. Could you use `document.activeElement` instead of `$(':focus')`
2. We don't use semicolons

File: jquery.pjax.js (hunk 2 above)
Comment (slimsag) [inline jquery.pjax.js]: I think that is out of scope of this change, personally. Should I go ahead and do it, though?

File: jquery.pjax.js (hunk 4 above)
Comment (slimsag) [inline jquery.pjax.js]: Done and done.

File: jquery.pjax.js (hunk 3 above)
Comment (slimsag) [inline jquery.pjax.js]: Done.

**Comment by mislav** [comment]
Thanks! Please squash all commits to one and force-push to this branch

**Comment by slimsag** [comment]
@mislav Done. PTAL.

**Comment by mislav** [comment]
Thanks!

**Comment by slimsag** [comment]
I really appreciate your timely responses @mislav; thanks for the hard work! :+1:
